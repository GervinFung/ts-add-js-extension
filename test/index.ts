import testParseConfig from './config';
import testTsAddJsExtension from './process';

const tests: ReadonlyArray<Readonly<[() => void, 'only'?]>> = [
    [testParseConfig],
    [testTsAddJsExtension],
];

const selectedTests = tests.filter(([_, only]) => only);

if (process.env.IS_CI && selectedTests.length) {
    throw new Error('cannot have "only" in ci cd');
}

(!selectedTests.length ? tests : selectedTests).forEach(([test]) => test());
