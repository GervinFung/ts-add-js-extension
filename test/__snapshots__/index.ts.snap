// Vitest Snapshot v1

exports[`ts add js extension > should read the code and ensure each import/export statemnt is properly formed for "1.js" 1`] = `
"import * as fs from 'fs';
import { fs } from './2.js';
import * as sample2 from './2.js';
import * as sampleExport2 from '../output/1.js';
import * as index from './index.js';
import * as index from './index.js';
export { index } from './index.js';
export * as index from './index.js';
export { default as x } from './index.js';
export { default as x } from './index.js';
"
`;

exports[`ts add js extension > should read the code and ensure each import/export statemnt is properly formed for "2.js" 1`] = `
"import { sample1 } from './1.js';
import './c.css';
export * from './1.js';
export * from '../output/1.js';
"
`;

exports[`ts add js extension > should read the code and ensure each import/export statemnt is properly formed for "3.js" 1`] = `
"import * as sampleExport2 from './2.js';
export * from './1.js';
export * as ays from './1.js';
"
`;

exports[`ts add js extension > should read the code and ensure each import/export statemnt is properly formed for "index.js" 1`] = `
"console.log('hi');
"
`;
